zypp/target/TargetImpl.cc

```diff
@@ -1352,7 +1352,9 @@ namespace zypp
       // Prepare execution of commit plugins:
       ///////////////////////////////////////////////////////////////////
       PluginExecutor commitPlugins;
-      if ( root() == "/" && ! policy_r.dryRun() )
+      const char *transactional_update = getenv( "TRANSACTIONAL_UPDATE" );
```

Comment:
``` 
namespace zypp::env {
  inline bool TRANSACTIONAL_UPDATE() 
  {
    return getenv( "TRANSACTIONAL_UPDATE" );
  }
} // namespace zypp::env
```
We try to collect environment vars in a dedicated `zypp::env` namespace (at the top of the file).
BTW is $TRANSACTIONAL_UPDATE reliably set throughout the whole SLE15-SPs? Using it in zypper would be smarter than the current detection.

Reply:
> is $TRANSACTIONAL_UPDATE reliably set throughout the whole SLE15-SPs?

Yes, it is

> Using it in zypper would be smarter than the current detection.

How this should look like ? Is zypper also responsible for deciding if the common plugins have to be run or not ?

Reply:
I have added your suggestion. Thank you !